src/locales/languages.ts

```diff
@@ -45,40 +46,15 @@ import zh_Hant from "./zh_Hant.json"
 //    see app/locales/pull.sh instructions
 // 3. import xy from `./xy.json` and add the language to the language block
 
-type Locale = string
-
-function toIETFLanguageTag(locale: Locale): Locale {
-  return locale.replace("_", "-").toLowerCase()
-}
-
-function getLanguageFromLocale(locale: Locale): Locale {
-  const [languageCode] = toIETFLanguageTag(locale).split("-")
-  return languageCode
-}
-
-async function setLocale(locale: Locale) {
-  dayjs.locale(toIETFLanguageTag(locale))
-  return await i18next.changeLanguage(locale)
-}
-
-type TextDirection = "ltr" | "rtl"
-
-export function useLanguageDirection(): TextDirection {
-  const { i18n } = useTranslation()
-  return i18n.dir()
-}
-
-export async function setUserLocaleOverride(locale: Locale): Promise<void> {
-  await setLocale(locale)
-  await StorageUtils.setUserLocaleOverride(locale)
-}
+type Resource = Record<Locale.Locale, ResourceLanguage>
 
 /* eslint-disable no-underscore-dangle */
-const AVAILABLE_TRANSLATIONS: Resource = {
+const ALL_LANGUAGE_RESOURCES: Record<Locale.Locale, ResourceLanguage> = {
```

Comment:
Based on the `type` above it seems like you may have wanted this to be `Resource`.